client/state/happychat/middleware.js

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_TRANSCRIPT_REQUEST,
+} from 'state/action-types';
+
+import { connection } from './common';
+
+import { receiveChatTranscript } from './actions';
+import { getHappychatTranscriptTimestamp } from './selectors';
+
+const debug = require( 'debug' )( 'calypso:happychat:actions' );
+
+export const requestTranscript = ( { getState, dispatch } ) => {
```

Comment:
Woooooo, I love this idea. At first I thought we might need something like `mock-socket` package, but you've achieved the same test coverage in a much lighter way ❤️